File: 476/main.py
class Solution:
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """

        # for example: 101(2) ^ 111(2) = 010(2)
        return num ^ int('1' * (len(bin(num)) - 2), 2) 

File: 476/main.c
#include <limits.h>

int findComplement(int num) {
    int mask = INT_MAX;
    while((num & mask) == num) {
        mask = mask >> 1;
    }
    if(mask != INT_MAX) {
        mask = (mask << 1) + 1;
    }

    return num ^ mask;
}
